### ISSUE TITLE
Bug: eslint ignores specific filename without reason

### ISSUE THREAD

**Issue by tombrouwer** [eslint/eslint#16834]
### Environment

Node version: v18.13.0
npm version: 8.19.2
Local ESLint version: 8.32.0
Global ESLint version: no
Operating System: WSL 2 (windows) Ubuntu 22.04

When I run eslint it skips a file even though there are errors in it, it doesn't display them. In debug mode it seems to check the file. And when I rename the file to another arbitrary name, eslint does show the errors.

Is is a typescript file en exports a simple abstract class.

The filename is: `AbstractDownloadRoute.ts` with this name it doesn't show errors.
When I change the filename to `BAbstractDownloadRoute.ts` (or something else) it shows the errors.
When I change the name back, it doesn't work anymore.

What could be the reason for this?

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "plugin:prettier/recommended"
  ],
  "parserOptions": {
    "ecmaVersion": 2022, 
    "sourceType": "module", 
    "project": "tsconfig.json"
  },
  "rules": {
    "@typescript-eslint/camelcase": ["off"],
    "@typescript-eslint/no-empty-function": ["off"],
    "@typescript-eslint/no-empty-interface": ["off"]
  }
}
```
</details>

```ts
import { AbstractRoute, RouteContract } from "{@name}/server";

export abstract class AbstractDownloadRoute extends AbstractRoute implements RouteContract {
  /** Route (parameter :fileName is needed for this class) */
    public static route = "/download/:fileId.:fileExtension";

  /** The requested file ID by parameter :fileId */
    public fileId: string;
}
```

### What did you expect to happen?

It will validate and fix always with all filenames.

### What actually happened?

It doesn't show or get erros for a specific filename.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
This is actually an issue with [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) rather than ESLint itself. Please open an issue there.
